### PR TITLE
add test command in docker

### PR DESCRIPTION
**Pull request by acgotaku**
- add fake database config
- add `npm run test` command

### CODE REVIEW

**Comment by acgotaku** [comment]
We have two database environment variables? such as:  `DB_MONGO_USER` `DB_PG_USER` ?

**Comment by lordfriend** [comment]
> We have two database environment variables? such as: `DB_MONGO_USER` `DB_PG_USER` ?

I mean you don't need to hard code all config values. a good example of this is a FakeConfigLoader which load() all config from environment variables but also set all its field public all other class directly override any field value. then you can hardcode some value in the spec file, such as db_port for different databases. I notice you have the same user and password for the two databases. so write two identical classes with only different hardcoded values is not optimal practice.

**Comment by acgotaku** [comment]
OK. The hard-coded values only in spec file.

File: src/storage/mongodb-store.spec.ts [hunk 1]
```diff
@@ -18,14 +17,13 @@ export class MongodbStoreSpec {
 
     @SetupFixture
     public setupFixture() {
-        if (!process.env.DB_NAME) {
```
Comment (lordfriend) [inline src/storage/mongodb-store.spec.ts]: please revert the warning message that requires the user to give a db_name

File: src/test/fake-config.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    private _mode: string;
+    private _dbMode: string
+    private _dbHost: string;
+    private _dbPort: number;
+    private _dbUser: string;
+    private _dbName: string;
+    private _dbPass: string;
+
+    public get mode(): string {
```
Comment (lordfriend) [inline src/test/fake-config.ts]: you could simply make the field public without explicitly using getter and setter.

**Pull request merged**